falcon-regression/merlin-core/src/main/java/org/apache/falcon/regression/core/util/NativeInstanceUtil.java

```diff
@@ -0,0 +1,176 @@
+            if (statusResult.getInstances() != null) {
+                break;
+            }
+            LOGGER.info(type + " " + entityName + " still doesn't have instance created");
+            TimeUtil.sleepSeconds(5);
+        }
+    }
+
+    /**
+     * Waits till given instance of process/feed reach expected state during specific time.
+     *
+     * @param cluster           ColoHelper - colo on which API to be executed
+     * @param entity            definition of entity which describes job
+     * @param instanceTime      time of instance
+     * @param expectedStatus    expected status we are waiting for
+     * @param frequency         frequency of process/feed
+     */
+    public static void waitTillInstanceReachState(ColoHelper cluster, Entity entity, String instanceTime,
+        CoordinatorAction.Status expectedStatus, Frequency frequency)
+        throws InterruptedException, IOException, AuthenticationException, URISyntaxException {
+        int totalMinutesToWait = InstanceUtil.getMinutesToWait(entity.getEntityType(), expectedStatus);
+        waitTillInstanceReachState(cluster, entity, instanceTime, expectedStatus, frequency, totalMinutesToWait);
+    }
+
+    /**
+     * Waits till given instance of process/feed reach expected state during
+     * specific time.
+     *
+     * @param cluster           ColoHelper - colo on which API to be executed
+     * @param entity            definition of entity which describes job
+     * @param instanceTime      time of instance
+     * @param expectedStatus    expected status we are waiting for
+     * @param frequency         frequency of process/feed
+     * @param totalMinutesToWait     total time(in minutes) to wait
+     */
+    public static void waitTillInstanceReachState(ColoHelper cluster, Entity entity, String instanceTime,
+        CoordinatorAction.Status expectedStatus, Frequency frequency, int totalMinutesToWait)
+        throws URISyntaxException, AuthenticationException, InterruptedException, IOException {
+        String entityName = entity.getName();
+        EntityType type = entity.getEntityType();
+
+        String endTime=getNextInstanceTime(instanceTime, frequency);
+        String params = "?start=" + instanceTime + "&end=" + endTime;
+
+        int maxTries = 50;
+        int totalSleepTime = totalMinutesToWait * 60;
+        int sleepTime = totalSleepTime / maxTries;
+        LOGGER.info(String.format("Sleep for %d seconds", sleepTime));
+        for (int i = 0; i < maxTries; i++) {
+            InstancesResult statusResult = cluster.getProcessHelper().getProcessInstanceStatus(entityName, params);
+            if (statusResult.getInstances() != null) {
+                if (statusResult.getInstances()[0].getStatus().name().equals(expectedStatus.toString())) {
+                    return;
+                }
+            }
+            LOGGER.info(type + " " + entityName + " still doesn't have expected status");
+            TimeUtil.sleepSeconds(5);
```

Comment:
TimeUtil.sleepSeconds(5) ?? 
I think you intended TimeUtil.sleepSeconds(sleepTime) ??